spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1430,6 +1430,22 @@ object QueryPlanSerde extends Logging with CometExprShim {
         val optExpr = scalarFunctionExprToProto("ascii", childExpr)
         optExprWithInfo(optExpr, expr, castExpr)
 
+      case StringDecode(binary, encoding) =>
+        encoding match {
+          case Literal(str, DataTypes.StringType) if str.toString == "utf-8" =>
```

Comment:
Should we use a case-insensitive comparison here?

Reply:
I'll check, but I expect Spark's front-end should have normalized this by the time it hits us.

Reply:
You could update the test to pass the string in a different case to confirm this?

Reply:
Yep, testing that now.

Reply:
Stunningly, Spark's parser is not to be trusted. It did not normalize it. I'll make the change. Good catch, and lesson learned for me.